### DISCUSSION TITLE
How can I set the visibility of images to be public or private based on user requests?

### DISCUSSION THREAD

**Discussion by 2020-byte** [supabase/supabase#29022] [Questions] [answered]
Is there a feature that allows moving files from a bucket where publicUrl is available to a bucket where publicUrl is not available? Or is there a way to achieve similar functionality? I want to configure images to be public or private based on user settings. When they are public, I want to use publicUrl, and when they are private, I want the publicUrl to be removed.

**Comment by GaryAustin1** [comment, marked as answer]
No.  There is not such feature as it is does on a bucket by bucket basis.  There is also no bucket to bucket move last time I checked.
You could make the file names hard to guess in a public bucket.  Then to hide them if you shared their public URL you would just rename them.

**Reply by 2020-byte** [reply]
I see, that sounds very simple yet guarantees powerful speed although there might be some security vulnerabilities. By the way, does the cacheControl value when uploading an file refer to the expiration time of the publicUrl cache? 
<img width="1285" alt="image" src="https://github.com/user-attachments/assets/46b5ca0a-f1ac-4e71-9bd9-18d315b44e90">

**Reply by GaryAustin1** [reply]
Not sure what a publicUrl cache is to you.
It sets how long the CDN and browsers will cache the file.

**Reply by 2020-byte** [reply]
Yes, how long a cached file can be used with a public URL without needing to retrieve the image from storage every time. Thank you!

**Reply by 2020-byte** [reply]
By the way, in Supabase, if you repeatedly try to access an incorrect path within the same bucket, is there any measure in place that blocks the IP from accessing the public URL for a certain period? Similar to how Facebook blocks access if there are multiple attempts to crawl the site in a short amount of time.
